src/platform/posix/posix_clock.c

```diff
@@ -19,22 +19,57 @@
 
 #ifndef NNG_USE_GETTIMEOFDAY
 
-// Use POSIX realtime stuff
-nni_time
-nni_clock(void)
+#if __APPLE__
+#include <mach/mach.h>
+#include <mach/mach_time.h>
+#endif
+
+// Use POSIX realtime stuff                                                          
+static inline nni_time via_clock_gettime(void)
 {
-	struct timespec ts;
-	nni_time        msec;
+ 	struct timespec ts;
+        nni_time        msec;
 
-	if (clock_gettime(NNG_USE_CLOCKID, &ts) != 0) {
-		// This should never ever occur.
+        if (clock_gettime(NNG_USE_CLOCKID, &ts) != 0) {
+                // This should never ever occur.                                     
 		nni_panic("clock_gettime failed: %s", strerror(errno));
-	}
+        }
 
 	msec = ts.tv_sec;
 	msec *= 1000;
-	msec += (ts.tv_nsec / 1000000);
-	return (msec);
+        msec += (ts.tv_nsec / 1000000);
+        return (msec);
+}
+
+nni_time
+nni_clock(void)
+{
+#if __APPLE__
```

Comment:
```suggestion
#if defined(__APPLE__)
```

you cannot use #if <IDENT> if the identifier could be undefined.

`__APPLE__` is defined only on apple platforms.

Reply:
Not sure what you mean... Try to compile this:
```
#if ZEC
#error ZEC is defined
#endif
```
and it will not report the compiler error. Yes it may look a little odd, but it's how people do it... maybe there's some case where someone does
```
#define __APPLE__ 0
```
which should be handled... I did not go into this.

Reply:
according to apple code snippets the correct check is

```
#ifdef __APPLE__
```
or
```
#if defined(__APPLE__)
```

as its something defined at the compiler level, not at source level. developers should not be re-defining things provided by the compiler. The compiler only states that it will be defined, not what its value will be.

Reply:
I've seen both used in various cross-platform codebases, `#if __APPLE__` seems "broader" as it handles the `#define __APPLE__ 0`, too. But, it's a simple change to go from one to the other, so, I don't care and it seems you do. :)